mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+                        "which is not fitted yet.\n"
+                        "Fit the estimator first, or change the "
+                        "cv argument value."
+                    ) from exc
+            return estimator
+
+    def _check_sample_weights(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+    ):
+        """
+        Validate sample_weight
+
+        Parameters
+        ----------
+        X: ArrayLike
+            Observed values.
+
+        y: ArrayLike
+            Target values.
+
+        sample_weight: Optional[NDArray] of shape (n_samples,)
+            Non-null sample weights.
+
+        Returns
+        -------
+        NDArray
+            The X observed values.
+
+        NDArray
+            the y target values.
+
+        Optional[NDArray] of shape (n_samples,)
+            Validated Non-null sample weights.
+
+        """
+        # Checking
+
+        X, y = indexable(X, y)
+        y = _check_y(y)
+        sample_weight, X, y = check_null_weight(sample_weight, X, y)
+
+        X = cast(NDArray, X)
+        y = cast(NDArray, y)
+        sample_weight = cast(Optional[NDArray], sample_weight)
+
+        return (
+            X, y,
+            sample_weight
+        )
+
+    def fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+        sample_weight: Optional[ArrayLike] = None,
+        groups: Optional[ArrayLike] = None,
+        **fit_params,
+    ) -> MapieCCPRegressor:
+        """
+        Fit the estimator.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y: ArrayLike of shape (n_samples,)
+            Training labels.
+
+        sample_weight: Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If ``None``, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+
+            By default ``None``.
+
+        groups: Optional[ArrayLike] of shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+            By default ``None``.
+
+        **fit_params: dict
+            Additional fit parameters for the estimator.
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+
+        """
+        self._check_fit_parameters()
+        if self.cv != 'prefit':
+            train_index = list(
+                cast(BaseCrossValidator, self.cv).split(X, y, groups)
+            )[0][0]
```

Comment:
Decompose it into two different instructions.

Reply:
Done